src/main/java/frc/robot/subsystems/DriveSub.java

```diff
@@ -0,0 +1,103 @@
+package frc.robot.subsystems;
+
+import com.ctre.phoenix6.hardware.TalonFX;
+
+import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.wpilibj.RobotBase;
+import edu.wpi.first.wpilibj.simulation.SingleJointedArmSim;
+import edu.wpi.first.wpilibj.smartdashboard.Mechanism2d;
+import edu.wpi.first.wpilibj.smartdashboard.MechanismLigament2d;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.constants.Constants;
+
+public class DriveSub extends SubsystemBase {
+
+    private TalonFX motor;
+    private SingleJointedArmSim armSim;
+    private Mechanism2d mech2d;
+    private MechanismLigament2d wheelLigament;
+    
+    // Constructor
+    public DriveSub(int motorID) {
+
+        motor = new TalonFX(motorID);
+
+        armSim = new SingleJointedArmSim(
+            DCMotor.getFalcon500(1), // assuming Falcon500 motor
```

Comment:
armSim should only be defined if the robot base isSimulation. You can choose whether to always define the mechanism or not.

Reply:
fixed it